notebook/services/kernels/kernelmanager.py

```diff
@@ -300,7 +301,8 @@ def shutdown_kernel(self, kernel_id, now=False):
             type=self._kernels[kernel_id].kernel_name
         ).dec()
 
-        return super(MappingKernelManager, self).shutdown_kernel(kernel_id, now=now)
+        ret = yield gen.maybe_future(super(MappingKernelManager, self).shutdown_kernel(kernel_id, now=now))
+        raise gen.Return(ret)
```

Comment:
This already supported async shutdown_kernel without any change, so I don't think we should merge this. Since `shutdown_kernel` is not itself async, returning the `super().shutdown_kernel()` works if shutdown_kernel is async, since it will return a Future or coroutine object, which gets passed through to the caller of this method.